src/core/backbone_router/local.cpp

```diff
@@ -238,6 +247,114 @@ void Local::UpdateBackboneRouterPrimary(Leader::State aState, const BackboneRout
+{
+    otError error = OT_ERROR_NONE;
+
+    VerifyOrExit(aPrefix.mLength > 0, error = OT_ERROR_INVALID_ARGS);
+
+    VerifyOrExit(mDomainPrefixConfig.mPrefix.mLength == aPrefix.mLength, error = OT_ERROR_NOT_FOUND);
+
+    VerifyOrExit(Ip6::Address::PrefixMatch(mDomainPrefixConfig.mPrefix.mPrefix.mFields.m8, aPrefix.mPrefix.mFields.m8,
+                                           BitVectorBytes(aPrefix.mLength)) >= aPrefix.mLength,
+                 error = OT_ERROR_NOT_FOUND);
+
+    if (IsEnabled())
+    {
+        error = Get<NetworkData::Local>().RemoveOnMeshPrefix(aPrefix.mPrefix.mFields.m8, aPrefix.mLength);
+        LogDomainPrefix("Remove", error);
+    }
+
+    mDomainPrefixConfig.mPrefix.mLength = 0;
+
+exit:
+    return error;
+}
+
+void Local::SetDomainPrefix(const NetworkData::OnMeshPrefixConfig &aConfig)
+{
+    otError error; // only used for logging.
+
+    if (IsEnabled() && mDomainPrefixConfig.mPrefix.mLength > 0)
+    {
+        error = Get<NetworkData::Local>().RemoveOnMeshPrefix(mDomainPrefixConfig.mPrefix.mPrefix.mFields.m8,
+                                                             mDomainPrefixConfig.mPrefix.mLength);
+
+        LogDomainPrefix("Remove", error);
+    }
+
+    mDomainPrefixConfig = aConfig;
+    LogDomainPrefix("Set", OT_ERROR_NONE);
+
+    if (IsEnabled())
+    {
+        error = Get<NetworkData::Local>().AddOnMeshPrefix(mDomainPrefixConfig);
+        LogDomainPrefix("Add", error);
+    }
+}
+
+void Local::RemoveDomainPrefixFromNetworkData(void)
+{
+    otError error = OT_ERROR_NOT_FOUND; // only used for logging.
+
+    if (mDomainPrefixConfig.mPrefix.mLength > 0)
+    {
+        error = Get<NetworkData::Local>().RemoveOnMeshPrefix(mDomainPrefixConfig.mPrefix.mPrefix.mFields.m8,
+
+                                                             mDomainPrefixConfig.mPrefix.mLength);
+        OT_ASSERT(error == OT_ERROR_NONE);
+    }
+
+    LogDomainPrefix("Remove", error);
+}
+
+void Local::AddDomainPrefixToNetworkData(void)
+{
+    otError error = OT_ERROR_NOT_FOUND; // only used for logging.
+
+    if (mDomainPrefixConfig.mPrefix.mLength > 0)
+    {
+        error = Get<NetworkData::Local>().AddOnMeshPrefix(mDomainPrefixConfig);
+    }
+
+    LogDomainPrefix("Add", error);
+}
+
+void Local::LogDomainPrefix(const char *aAction, otError aError)
+{
+    if (mDomainPrefixConfig.mPrefix.mLength > 0)
+    {
+        otLogDebgNetData("prefix: %s/%d",
+                         (*static_cast<Ip6::Address *>(&mDomainPrefixConfig.mPrefix.mPrefix)).ToString().AsCString(),
+                         mDomainPrefixConfig.mPrefix.mLength);
+    }
+    else
+    {
+        otLogInfoNetData("No available prefix");
+    }
+
+    otLogInfoNetData("%s Domain Prefix: %s%s", aAction, otThreadErrorToString(aError));
+}
+
+void Local::LogBackboneRouterService(const char *aAction, otError aError)
+{
+    otLogDebgNetData("local BBR: seqno (%d), delay (%ds), timeout (%ds)", mSequenceNumber, mReregistrationDelay,
```

Comment:
Need new region: `otLogDebgBbr` ?

Reply:
planned to add BBR region log in a separate PR, let's keep this one simple for now?

Reply:
I am fine with another PR.